src/IB/IBHydrodynamicSurfaceForceEvaluator.cpp

```diff
@@ -70,6 +68,31 @@ sign(const double X)
 {
     return ((X > 0) ? 1 : ((X < 0) ? -1 : 0));
 } // sign
+
+inline void
+get_physical_coordinate(IBTK::Vector3d& side_coord, Pointer<Patch<NDIM> > patch, const SideIndex<NDIM>& side_idx)
+{
+    const int axis = side_idx.getAxis();
+    Pointer<CartesianPatchGeometry<NDIM> > patch_geom = patch->getPatchGeometry();
+    const double* patch_X_lower = patch_geom->getXLower();
+    const Box<NDIM>& patch_box = patch->getBox();
+    const Index<NDIM>& patch_lower_idx = patch_box.lower();
+    const double* const patch_dx = patch_geom->getDx();
+
+    for (int d = 0; d < NDIM; ++d)
+    {
+        if (d == axis)
+        {
+            side_coord[d] = patch_X_lower[d] + patch_dx[d] * (static_cast<double>(side_idx(d) - patch_lower_idx(d)));
+        }
+        else
+        {
+            side_coord[d] =
+                patch_X_lower[d] + patch_dx[d] * (static_cast<double>(side_idx(d) - patch_lower_idx(d)) + 0.5);
+        }
+    }
+    return;
+} // get_physical_coordinate
```

Comment:
Do we already have a function that does this somewhere, like in `IndexUtilities`?

Reply:
(I added this to IndexUtilities)